esp-rtos/src/esp_radio/timer_queue.rs

```diff
@@ -300,17 +300,25 @@ impl TimerImplementation for Timer {
         TIMER_QUEUE.inner.with(|q| {
             let timer = unsafe { Box::from_raw(timer.cast::<Timer>().as_ptr()) };
 
-            // There are two cases:
-            // - We can remove the timer from the queue - we can drop it.
-            // - We can't remove the timer from the queue. There are the following cases:
-            //   - The timer isn't in the queue. We can drop it.
-            //   - The timer is in the queue and the queue is being processed. We need to mark it to
-            //     be dropped by the timer queue.
-            if q.dequeue(&timer) {
-                core::mem::drop(timer);
-            } else {
+            // There are multiple cases:
+            // - the queue is currently processing - mem-forget the timer and mark it to be dropped
+            //   later in during processing.
+            //
+            // - the queue is currently not processing
+            //      - we can dequeue the timer -> drop it
+            //      - we can't dequeue the timer (i.e. it's not in the queue) -> drop it
+            if q.processing {
```

Comment:
This isn't quite correct yet, I think. There is a chance the timer has already been processed, and it may have become inactive, meaning it was not enqueued again. Note that `fn process` does not hold the mutex throughout the whole process, it takes it twice per timer.

I am not sure, but I think we do NOT have enough information currently to tell if a timer has been processed fully, but did not get re-enqueued, and we need to drop it. We also must not drop the timer while its callback is running. We need some way to detect if the timer item has already left the second critical section of its processing.